.github/workflows/ci.yml

```
@@ -0,0 +1,288 @@
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go }}
+          cache: true
+      
+      - name: Unit Tests
+        run: |
+          go test -v -race -coverprofile=coverage.txt -covermode=atomic ./src/...
+      
+      - name: Upload Coverage
+        if: matrix.os == 'ubuntu-latest'
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)